hibernate-core/src/main/java/org/hibernate/boot/model/internal/AnnotatedJoinColumn.java

```diff
@@ -109,9 +110,13 @@ static AnnotatedJoinColumn buildJoinColumn(
 			String defaultColumnSuffix) {
 		if ( joinColumn != null ) {
 			if ( mappedBy != null ) {
-				throw new AnnotationException( "Association '"
-						+ getRelativePath( propertyHolder, inferredData.getPropertyName() )
-						+ "' is 'mappedBy' a different entity and may not explicitly specify the '@JoinColumn'" );
+				throw new AnnotationException(
+						String.format(
+								Locale.ROOT,
+								"Association '%s' is 'mappedBy=%s' a different entity and may not explicitly specify the '@JoinColumn'",
```

Comment:
Folks, I definitely appreciate the desire to improve error messages, but I think this change makes the message worse:

1.  `... is 'mappedBy=%s' a different entity` doesn't read like a well-formed English sentence, and it's not even correct Java (it has missing quotes). If you really think it's important to include the name of the mapping attribute, it should be `... is 'mappedBy' the attribute '%s' of a different entity`.
2. If I'm not mistaken, I used `getRelativePath()` here very deliberately because I didn't want a long-ass qualified name right at the start of the exception message. That really harms readability of the message. When we include full-qualified names, we usually try to put them at the end of the error message.

Reply:
Perhaps:

    "Association '%s' of entity '%s' is 'mappedBy' a different entity and may not explicitly specify the '@JoinColumn'"

where the second `%s` is the (unqualified) entity name.